zombies.js

```diff
@@ -79,6 +96,101 @@
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
 
+class Player {
+  constructor(name, health, strength, speed) {
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this._pack = [];
+    this._maxHealth = health;
+    this.equipped = false;
+  }
+
+  getPack() {
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  checkPack() {
+    console.log(Player.getPack);
+    return;
+  }
+
+  takeItem(item) {
+    if (this._pack.length < 3) {
+      console.log(`${this.name} acquired ${item.name}`);
+      this._pack.push(item);
+      return true;
+    } else {
+      console.log("Pack is full so the item could not be stored.");
+      return false;
+    }
+  }
+
+  discardItem(item) {
+    let idx = this.getPack().indexOf(item);
+    if (idx === -1) {
+      console.log("Nothing was discarded since the item could not be found.");
+      return false;
+    } else if (idx >= 0) {
+      this.getPack().splice(idx, 1);
+      console.log(`${this.name} has discarded ${item.name}`);
+      return true;
+    }
+  }
+
+  equip(itemToEquip) {
+    let weaponIdx = this.getPack().indexOf(itemToEquip);
+    let isWeapon = itemToEquip instanceof Weapon;
+
+    if (weaponIdx >= 0 && isWeapon) {
+      if (this.equipped === false) {
+        this.equipped = itemToEquip;
+        this.getPack().splice(weaponIdx, 1);
+      } else {
+        this.getPack().push(this.equipped);
+        this.equipped = itemToEquip;
+        this.getPack().splice(weaponIdx, 1);
```

Comment:
This should pull the itemToEquip out of the pack first, then put the equipped item into the pack, then equip the itemToEquip